src/index.ts

```diff
@@ -109,3 +109,49 @@ export const compare = memoize((versionA: string | null, versionB: string | null
 	}
 	return versionA.localeCompare(versionB);
 }, (a: string, b: string) => `${a} && ${b}`);
+
+/**
+ * @summary Return the major version number
+ * @name compare
+ * @public
+ * @function
+ *
+ * @description Returns the major version number in a semver string.
+ * If the presented version is a falsey value, it returns `0`. If the version is
+ * not a valid semver string, it returns the first number it finds in the string.
+ * If there are no numbers in the provided string, it returns `1`.
+ *
+ * @param {string|null} version - The version string to evaluate
+ *
+ * @returns {number} - The major version number
+ *
+ * @example
+ * resinSemver.major(null); //--> 0
+ *
+ * resinSemver.major('Resin OS v2.0.5'); //--> 2
+ *
+ * resinSemver.major('Resin OS v2.0.5'); //--> 2
+ *
+ * resinSemver.major('Linux 14.04'); //--> 14
+ *
+ * resinSemver.major('My development version'); //--> 1
```

Comment:
This logic is where we start getting inconsistencies. It's debatable how they should work out, but we should have some coherent answers. Intuitively, I want to assume that if compare says A > B, then major(A) >= major(B) should be true. I.e. if one version is newer than another, then I should be able to assume it's the same or a later major version.

If we want to make that assumption hold, then these examples don't work (v0.0.1 compares above My development version, but has a lower major result).

The same holds for strings that happen to have numbers in them (Linux 14.04): we should treat them similarly across all the methods here. That means if we're going to optimistically try to infer the version number from them here, then we should use that same inferred number in compare.

Totally happy to debate what the rules for this to be, but they should be consistent.

Personally: I'd do proper semver comparison and major/minor/etc parsing only on valid semver strings plus a couple of clear specific semver-compatible equivalents (e.g. dropping Resin OS v prefixes, fixing specific not-quite-semver suffixes like .dev), and for everything else compare them below, and return null for major/minor/etc. Essentially we do comparison on the set of versions we're really confident about, and we treat everything else as equivalent 'other'. How does that sound?